common/app/common/dfp/PageskinAdAgent.scala

```diff
@@ -55,9 +55,15 @@ trait PageskinAdAgent {
   }
 
   // The ad unit is considered to have a page skin if it has a corresponding sponsorship.
-  def hasPageSkin(fullAdUnitPath: String, metaData: MetaData, edition: Edition): Boolean = {
+  // If the sponsorship is targetting an adtest we also consider that the request URL includes the same adtest param
+  def hasPageSkin(fullAdUnitPath: String, metaData: MetaData, edition: Edition, adTestParam: Option[String]): Boolean = {
```

Comment:
I think `hasPageSkin` should only test whether `sponsorship.targetsAdTest` is empty. This function would only return true if there is a **Live page skin.**

The logic for whether a page skin applies with ad tests should be in `hasPageSkinOrAdTestPageSkin`. Otherwise, these functions were identical, and taking the exact same parameters.

```suggestion
  def hasPageSkin(fullAdUnitPath: String, metaData: MetaData, edition: Edition): Boolean = {
```

Furthermore, there might be an opportunity to get the request implicitly in `hasPageSkinOrAdTestPageSkin`, but I'm not too familiar with the syntax.

Reply:
- `hasPageSkin` is responsible for the css class `has-page-skin` on the `body` element of the page  https://github.com/guardian/frontend/blob/5adaf9bab30be9e61a0e3fc87d322854a39291c5/common/app/views/main.scala.html#L34
^ This is the one what actually shrinks the page without the adtest param present on the URL 

- `hasPageSkinOrAdTestPageSkin` is responsible  for adding the slot `pageskin-inread` on the page 
https://github.com/guardian/frontend/blob/5adaf9bab30be9e61a0e3fc87d322854a39291c5/common/app/views/main.scala.html#L45

I think an improvement will be to merge the 2 functions into 1 since they are identical anw

Reply:
👍